src/kOS/Utilities/VesselUtils.cs

```diff
@@ -318,7 +389,312 @@ public static object GetSolarPanelStatus(Vessel vessel)
 
         public static void SolarPanelCtrl(Vessel vessel, bool state)
         {
-            vessel.rootPart.SendEvent(state ? "Extend" : "Retract");
+            foreach (var p in vessel.parts)
+            {
+                foreach (var c in p.FindModulesImplementing<ModuleDeployableSolarPanel>())
+                {
+                    if (state) { c.Extend(); }else { c.Retract(); }
+                }
+            }
+        }
+
+        public static object GetRadiatorStatus(Vessel vessel)
+        {
+            var atLeastOneRadiator = false; // No radiators at all? Always return false
+
+            foreach (var p in vessel.parts)
+            {
+                foreach (var c in p.FindModulesImplementing<ModuleActiveRadiator>())
+                {
+                    atLeastOneRadiator = true;
+
+                    if (! c.IsCooling)
```

Comment:
Should we also check for `!p.ShieldedFromAirstream` so that this will not always return false if you have a radiator stowed inside of a cargo bay.  I believe that parts which are shielded will automatically not extend if given the command.